src/main/java/ai/lucidtech/las/sdk/Client.java

```diff
@@ -45,315 +44,1079 @@ public Client(Credentials credentials) {
+     * @return Secret response from API
+     * @throws IOException General IOException
+     * @throws APIException Raised when API returns an erroneous status code
+     * @throws MissingAccessTokenException Raised if access token cannot be obtained
+     */
+    public JSONObject createSecret(Map<String, String> data)
+    throws IOException, APIException, MissingAccessTokenException {
+        return this.createSecret(data, null);
+    }
+
+    /**
+     * Creates a secret, calls the POST /secrets endpoint.
+
+     * @param data Key-Value pairs to store secretly
+     * @return Secret response from API
+     * @throws IOException General IOException
+     * @throws APIException Raised when API returns an erroneous status code
+     * @throws MissingAccessTokenException Raised if access token cannot be obtained
+     */
+    public JSONObject createSecret(JSONObject data)
+    throws IOException, APIException, MissingAccessTokenException {
+        return this.createSecret(data, null);
+    }
+
+    /**
+     *  List secrets available, calls the GET /secrets endpoint.
      *
-     * @param documentId The document id to run inference and create a prediction on
-     * @return response from the API
+     * @see ListSecretsOptions
+     * @param options Additional options to pass along as query parameters
+     * @return Secrets response from REST API
      * @throws IOException General IOException
      * @throws APIException Raised when API returns an erroneous status code
      * @throws MissingAccessTokenException Raised if access token cannot be obtained
      */
-    public JSONObject getDocument(String documentId) throws IOException, APIException, MissingAccessTokenException {
-        HttpUriRequest request = this.createAuthorizedRequest("GET", "/documents/" + documentId);
+    public JSONObject listSecrets(ListSecretsOptions options)
+    throws IOException, APIException, MissingAccessTokenException {
+        HttpUriRequest request = this.createAuthorizedRequest("GET", "/secrets", options.toList());
         String response = this.executeRequest(request);
         return new JSONObject(response);
     }
 
     /**
+     *  List secrets available, calls the GET /secrets endpoint.
      *
-     * @return All documents from REST API
+     * @return Secrets response from REST API
      * @throws IOException General IOException
      * @throws APIException Raised when API returns an erroneous status code
      * @throws MissingAccessTokenException Raised if access token cannot be obtained
      */
-    public JSONObject listDocuments() throws IOException, APIException, MissingAccessTokenException {
-        HttpUriRequest request = this.createAuthorizedRequest("GET", "/documents");
-        String response = this.executeRequest(request);
-        return new JSONObject(response);
+    public JSONObject listSecrets() throws IOException, APIException, MissingAccessTokenException {
+        return this.listSecrets(new ListSecretsOptions());
```

Comment:
Should we not send null here?

Reply:
We don't handle options=null in the other class.

Reply:
but will fix that soon